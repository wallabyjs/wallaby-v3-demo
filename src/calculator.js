export function add(a, b) {
  return a + b;
}

export function multiply(a, b) {
  return a * b;
}

export function slowFibonacci(n) {
  if (n > 40) {
    throw new Error('too ambitious; please provide a number <= 40');
  }
  return (n <= 1) ? n : slowFibonacci(n - 1) + slowFibonacci(n - 2);
}