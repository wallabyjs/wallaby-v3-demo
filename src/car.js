export class Car {
  constructor(initialSpeed) {
    this.engine = '2.85 L';
    this.power = '97 kW';
    this.speed = initialSpeed;
  }

  accelerate() {
    this.speed++;
  }

  brake() {
    // fix this reference to get rid of the associated failure(s)
    speed--;
  }
}
