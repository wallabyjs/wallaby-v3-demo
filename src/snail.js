export class Snail {
  async move() {
    console.log('this might take a while...');
    const journeyStart = Date.now();
    await new Promise((r) => {
      setTimeout(() => r(), 4500);
    });
    console.log(`moved for ${Date.now() - journeyStart}ms`);
  }
}