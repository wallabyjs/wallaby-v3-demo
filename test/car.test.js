import { describe, it, expect, beforeEach } from 'vitest';
import { Car } from '../src/car';

let car;

describe('car', () => {
  beforeEach(() => {
    car = new Car(88);
  });

  it('has correct engine', () => {
    expect(car.engine).toBe('2.85 L');
  });

  it('accelerates', () => {
    car.accelerate();
    expect(car).toMatchSnapshot();
  });

  it('brakes', () => {
    car.brake();
    expect(car.speed).toBe(87);
  });
});
