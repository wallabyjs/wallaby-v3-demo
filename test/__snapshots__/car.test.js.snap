// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`car > accelerates 1`] = `
Car {
  "engine": "2.85 L",
  "power": "97 kW",
  "speed": 89,
}
`;
